src/protocolhandler.ts

```diff
@@ -357,11 +359,16 @@ export class ProtocolHandler extends EventEmitter {
     }
 
     private connect(): Promise<Transport> {
+        this.currentServer.lastConnect = Date.now();
         this.prepareConnection();
-        if (this.currentServer.didConnect) {
+        if (this.reconnecting) {
+            this.currentServer.reconnects += 1;
             this.client.emit('reconnecting', this.url.href);
+        } else {
+            // not on the 'reconnect' loop but honoring the reconnect policy
+            this.currentServer.connects += 1;
```

Comment:
This is `.connects` and you comment about reconnect policy. Is this for the initial connect that follows the same logic than reconnect in term of attempts, etc..?

Reply:
this is left behind, I have removed them (was trying to reuse the reconnect logic), but initial connect logic either retries forever (`awaitFirstConnect`) or gives up after all servers are tried and failed.

Reply:
Oh, I thought that the Node client would optionally use the reconnect config also for the initial connect..

Reply:
no it was an in-between strat that I dropped, but didn't remove the counter.